core/src/main/java/org/jdbi/v3/core/mapper/EnumByNameMapperFactory.java

```diff
@@ -22,7 +22,11 @@
 
 /**
  * Produces enum column mappers, which map enums from varchar columns using {@link Enum#valueOf(Class, String)}.
+ *
+ * @deprecated there is no reason for this to be API
  */
+@Deprecated
+// TODO jdbi4: delete
```

Comment:
The deprecation message is inaccurate. This and `EnumByOrdinalMapperFactory` were made public API to enable changing the default enum strategy via a registration:

Core:

```
jdbi.registerColumnMapper(EnumByOrdinalMapperFactory.class);
```

SQL Object:

```
@RegisterColumnMapperFactory(EnumByOrdinalMapperFactory.class)
```

With that said, we never added an equivalent for `@RegisterArgumentFactory`, so it wasn't a full solution.

Let's change the deprecation message to redirect users to the new API options that replace this class: the config class `default*` property, or the annotations.